### ISSUE TITLE
Print function for DiscreteVectorLearnerParam with unknown length is broken

### ISSUE THREAD

**Issue by mb706** [berndbischl/ParamHelpers#180]
```R
> x =  makeDiscreteVectorLearnerParam("test", default = list(), values = c("a", "b", "c"), len = NA)
> x
Error in if (par$type == "discretevector" && length(x) != par$len) stopf("Length of x must be %i!",  : 
  missing value where TRUE/FALSE needed
```

**Comment by berndbischl** [comment]
the problem seems to be your default? and should that even be valid?

**Comment by berndbischl** [comment]
if you remove the default=list() it works

**Comment by mb706** [comment]
So LearnerParams with unknown vector length should not have a default?

**Comment by berndbischl** [comment]
this also breaks 

x =  makeDiscreteVectorLearnerParam("test", default = list("a", "b"), values = c("a", "b", "c"), len = NA)

**Comment by berndbischl** [comment]
> So LearnerParams with unknown vector length should not have a default?

no, i am not saying this. i was just wondering wether the list() is a valid defaults. surely vecparams can have defaults. i was wondering whether the problems comes from the edge case of a default where the vector is length 0.

but apparantly other defaults have the same problem

**Comment by berndbischl** [comment]
the method that really breaks in the background is this

> x =  makeDiscreteVectorLearnerParam("test", values = c("a", "b", "c"), len = NA)
> paramValueToString(x, list())
Error in if (par$type == "discretevector" && length(x) != par$len) stopf("Length of x must be %i!",  : 
  missing value where TRUE/FALSE needed
> paramValueToString(x, list("a"))
Error in if (par$type == "discretevector" && length(x) != par$len) stopf("Length of x must be %i!",  : 
  missing value where TRUE/FALSE needed

**Comment by mb706** [comment]
Also
```R
> x= makeNumericVectorLearnerParam("test", default = numeric(0), lower = 0, upper = 10, len = NA)
> x
           Type len        Def  Constr Req Tunable Trafo
1 numericvector  NA numeric(0) 0 to 10   -    TRUE     -
Used: train.
```
works (and I think length 0 vectors should be possible imho).

**Comment by berndbischl** [comment]
> (and I think length 0 vectors should be possible imho).

i agree. and the problem is as i posted above something else anyway. but of course we can then unit test your case with 0 length too
